### ISSUE TITLE
[external bug report][from claude] Multi-tile reduce dims=[0] duplicates first tile-row result

### ISSUE THREAD

**Issue by zoecarver** [tenstorrent/tt-lang#381]
## Summary

`reduce_sum(input, scaler, dims=[0])` on a multi-tile block only computes the reduction for the first tile-row, then duplicates that result to all subsequent tile-rows.

This is against PR #370, which introduces the new reduce implementation. This functionality is not yet on main.

## Reproducer

```python
import torch
import ttnn
import ttl

@ttl.kernel(grid=(1, 1))
def reduce_sum_row_2x2_kernel(inp, scaler, out):
    inp_dfb = ttl.make_dataflow_buffer_like(inp, shape=(2, 2), buffer_factor=2)
    scaler_dfb = ttl.make_dataflow_buffer_like(scaler, shape=(1, 1), buffer_factor=2)
    out_dfb = ttl.make_dataflow_buffer_like(out, shape=(2, 1), buffer_factor=2)

    @ttl.compute()
    def compute_fn():
        with inp_dfb.wait() as x, scaler_dfb.wait() as s, out_dfb.reserve() as o:
            o.store(ttl.math.reduce_sum(x, s, dims=[0]))

    @ttl.datamovement()
    def dm_read():
        with inp_dfb.reserve() as blk:
            tx = ttl.copy(inp[0:2, 0:2], blk)
            tx.wait()
        with scaler_dfb.reserve() as blk:
            tx = ttl.copy(scaler[0, 0], blk)
            tx.wait()

    @ttl.datamovement()
    def dm_write():
        with out_dfb.wait() as blk:
            tx = ttl.copy(blk, out[0:2, 0:1])
            tx.wait()

if __name__ == "__main__":
    device = ttnn.open_device(device_id=0)

    # Distinct values per tile quadrant to expose the bug
    inp_t = torch.zeros((64, 64), dtype=torch.bfloat16)
    inp_t[:32, :32] = 1.0   # tile [0,0]
    inp_t[:32, 32:] = 2.0   # tile [0,1]
    inp_t[32:, :32] = 3.0   # tile [1,0]
    inp_t[32:, 32:] = 4.0   # tile [1,1]

    scaler_t = torch.ones((32, 32), dtype=torch.bfloat16)

    inp = ttnn.from_torch(inp_t, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                          device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG)
    scaler = ttnn.from_torch(scaler_t, dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                             device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG)
    out = ttnn.from_torch(torch.zeros((64, 32), dtype=torch.bfloat16),
                          dtype=ttnn.bfloat16, layout=ttnn.TILE_LAYOUT,
                          device=device, memory_config=ttnn.DRAM_MEMORY_CONFIG)

    reduce_sum_row_2x2_kernel(inp, scaler, out)
    result = ttnn.to_torch(out)

    # dims=[0] keeps rows, sums across columns:
    #   row 0: 1.0*32 + 2.0*32 = 96
    #   row 1: 3.0*32 + 4.0*32 = 224
    print(f"row0: {result[0,0].item()}")   # Got: 96.0 (correct)
    print(f"row1: {result[32,0].item()}")  # Got: 96.0 (WRONG, expected 224.0)
    ttnn.close_device(device)
```

## Expected vs actual

- **Row 0**: 96.0 (correct)
- **Row 1**: 96.0 (wrong -- should be 224.0, duplicates row 0)

## Note on existing test coverage

The existing `test_reduce_sum_row_2x2` in `test/python/test_reduce_ops.py` uses all-ones input, which makes this bug invisible since every tile-row produces the same sum (64.0). The test also only checks indices within tile-row 0 (`result[0, 0]` and `result[15, 0]`), never tile-row 1 (`result[32, 0]`). The test should be updated to use distinct values per tile and check across tile boundaries.

## Environment

- HW only
- `dims=[1]` and `dims=[0,1]` work correctly
- Tested with 2x2 tile grid (64x64 input)

**Comment by zoecarver** [comment]
fixed in #289

**Comment by brnorris03** [comment]
Fixed by #597 (e72ba5d00). The numeric-scaler form changed in #589, and the multi-output-tile reduce bug (same shape class as here -- a reduced dim leaving more than one output tile) was fixed in #597. Current coverage in `test/python/test_reduce.py` exercises multi-tile reduce with distinct `torch.rand` values on 2x2/4x4/non-square grids for dim0/dim1/both and checks the entire output against a torch golden (`_expected_reduce_tensor` + `assert_allclose`, `test_reduce.py:572`), which would catch tile-row duplication. Closing as fixed; the all-ones-masking concern is addressed by the `_random` variants.
